File: 0189-rotate-array/0189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        if len(nums)==1:
            return nums
        L=[]
        for i in range(len(nums)):
            L.append(nums[i])
        for i in range(len(nums)):
            pos=i
            new_pos=i+k
            if new_pos<len(nums):
                nums[new_pos]=L[i] 
            else:
                while new_pos>=len(nums):
                    new_pos=new_pos-len(nums)
                nums[new_pos]=L[i]
        """
        Do not return anything, modify nums in-place instead.
        """
        

File: 0189-rotate-array/0189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k=nums.size()-(k%nums.size());
        int temp[k];
        for(int i=0;i<k;i++){
            temp[i]=nums[i];
        }
        for(int i=0;i<nums.size()-k;i++){
            nums[i]=nums[k+i];
        }
        int j=0;
        for(int i=nums.size()-k;i<nums.size();i++){
            nums[i]=temp[j];
            j++;
        }
    }
};